include/deal.II/distributed/fully_distributed_tria.h

```diff
@@ -0,0 +1,406 @@
+     *
+     * @author Peter Munch, 2019
+     */
+    template <int dim>
+    struct CellInfo
+    {
+      /**
+       * Constructor.
+       */
+      CellInfo() = default;
+
+      /**
+       * Unique CellID of the cell.
+       */
+      CellId::binary_type id;
+
+      /**
+       * subdomain_id of the cell.
+       */
+      types::subdomain_id subdomain_id;
+
+      /**
+       * level_subdomain_id of the cell.
+       */
+      types::subdomain_id level_subdomain_id;
+
+      /**
+       * Manifold id of the cell.
+       */
+      types::material_id manifold_id;
+
+      /**
+       * Manifold id of all lines of the cell.
+       *
+       * @note Only used for 2D and 3D.
+       */
+      std::array<types::material_id, GeometryInfo<dim>::lines_per_cell>
+        manifold_line_ids;
+
+      /**
+       * Manifold id of all faces of the cell.
+       *
+       * @note Only used for 3D.
+       */
+      std::array<types::material_id, GeometryInfo<dim>::quads_per_cell>
+        manifold_quad_ids;
+
+      /**
+       * List of face number and boundary id of all non-internal faces of the
+       * cell.
+       */
+      std::vector<std::pair<unsigned int, types::boundary_id>> boundary_ids;
+    };
+
+
+
+    /**
+     * Data used to construct a parallel::fullydistributed::Triangulation.
+     *
+     * @author Peter Munch, 2019
+     */
+    template <int dim, int spacedim>
+    struct ConstructionData
+    {
+      /**
+       * Cells of the locally-relevant coarse-grid triangulation.
+       */
+      std::vector<CellData<dim>> coarse_cells;
+
+      /**
+       * Vertices of the locally-relevant coarse-grid triangulation.
+       */
+      std::vector<Point<spacedim>> coarse_cell_vertices;
+
+      /**
+       * Mapping from coarse-grid index to coarse-grid id.
+       */
+      std::vector<types::coarse_cell_id> coarse_cell_index_to_coarse_cell_id;
+
+      /**
+       * CellInfo for each locally relevant cell on each level.
+       */
+      std::vector<std::vector<CellInfo<dim>>> cell_infos;
+    };
+
+
+    /**
+     * A distributed triangulation with a distributed coarse grid.
+     *
+     * The motivation for parallel::fullydistributed::Triangulation has its
+     * origins in the following observations about complex geometries and/or
+     * about given meshes created by an external mesh generator. We regard
+     * complex geometries as geometries that can be meshed only with a
+     * non-negligible number of coarse cells (>10,000):
+     * - storing the coarse-grid information on every process is too expensive
+     *   from a memory point of view (as done by
+     *   parallel::distributed::Triangulation). Normally, a process only needs a
+     *   small section of the global triangulation, i.e., a small section of the
+     *   coarse grid such that a partitioning of the coarse grid is indeed
+     *   essential. We call cells needed by a process locally relevant cells.
```

Comment:
Is this different from "locally owned"? I'm asking because that's a well established term in deal.II. It may be useful to make clear whether the "locally relevent cells" require an overlap between processors. You might want to look to the corresponding glossary entries.

Reply:
below it is described as:
```cpp
     * - a locally relevant coarse-grid triangulation
     *   (vertices, cell definition; including a layer of ghost cells)
```